packages/cli/src/cog/cog.job.ts

```diff
@@ -0,0 +1,340 @@
+ */
+export function extractYearRangeFromName(name: string): [number, number] {
+    const re = /(?:^|\D)(\d{4})(?:-(\d{2}))?(?:$|\D)/g;
+
+    const years: number[] = [];
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        years.push(parseInt(m[1]));
+        if (m[2] != null) years.push(parseInt(m[1].slice(0, 2) + m[2]));
+    }
+
+    if (years.length == 0) {
+        return [-1, -1];
+    }
+
+    years.sort();
+
+    return [years[0], years[years.length - 1] + 1];
+}
+
+export class CogJob {
+    stac: CogStac;
+    private cacheTargetZoom?: {
+        gsd: number;
+        zoom: number;
+    };
+
+    /**
+     * The Name to used when just producing one cog to cover the full extent
+     */
+    static OneCogName = '0-0-0';
+
+    static async build({
+        id,
+        imageryName,
+        metadata,
+        ctx,
+        srcPoly,
+        addAlpha,
+    }: {
+        id: string;
+        imageryName: string;
+        metadata: CogBuilderMetadata;
+        ctx: JobCreationContext;
+        srcPoly: MultiPolygon;
+        addAlpha: boolean;
+    }): Promise<CogJob> {
+        const nowStr = new Date().toISOString();
+
+        const coordinates = multiPolygonToWgs84(srcPoly, Projection.get(metadata.projection).toWgs84);
+        const bbox = Wgs84.multiPolygonToBbox(coordinates);
+
+        const sourceFs = FileOperator.create(ctx.sourceLocation);
+
+        let description: string | undefined;
+        const providers: StacProvider[] = [];
+        const links: StacLink[] = [
+            {
+                href: FileOperator.join(ctx.outputLocation.path, 'collection.json'),
+                type: 'application/json',
+                rel: 'self',
+            },
+        ];
+        let sourceStac = {} as StacCollection;
+
+        const interval: [string, string][] = [];
+        try {
+            const sourceCollectionPath = FileOperator.join(ctx.sourceLocation.path, 'collection.json');
+            sourceStac = JSON.parse((await sourceFs.read(sourceCollectionPath)).toString()) as StacCollection;
+            description = sourceStac.description;
+            interval.push(...(sourceStac.extent?.temporal?.interval ?? []));
+            links.push({ href: sourceCollectionPath, rel: 'sourceImagery', type: 'application/json' });
+            if (sourceStac.providers != null) providers.push(...sourceStac.providers);
+        } catch (err) {
+            if (!sourceFs.isNotFoundError(err)) {
+                throw err;
+            }
+        }
+        const keywords = sourceStac.keywords ?? CogStacKeywords.slice();
+        const license = sourceStac.license ?? CogStacLicense;
+        const title = sourceStac.title ?? titleizeImageryName(imageryName);
+
+        if (interval.length == 0) {
+            const years = extractYearRangeFromName(imageryName);
+            if (years[0] != -1) {
+                interval.push(years.map((y) => `${y}-01-01T00:00:00Z`) as [string, string]);
```

Comment:
do these strings, need to be full ISO date strings, or could we just do years? 

cause most of the flying is not done between jan 1st -> jan 1st? aka we are making up data that is wrong

Reply:
This is the format we have to work with. This is what LSD are doing.